### PR TITLE
nfts: mints

### PR DESCRIPTION
**Pull request by skylarbarrera**
## What changed (plus any additional context for devs)
adds in app minting functionality to mint.fun collections 

## What to test

minting nfts daaawg

### CODE REVIEW

File: src/resources/reservoir/mints.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import Routes from '@/navigation/routesNames';
+import { IS_DEV } from '@/env';
+import { logger } from '@/logger';
+import { WrappedAlert as Alert } from '@/helpers/alert';
+import * as lang from '@/languages';
+
+const client = IS_DEV ? arcDevClient : arcClient;
+const showAlert = () => {
+  Alert.alert(
+    'Could not find collection',
```
Comment (skylarbarrera) [inline src/resources/reservoir/mints.ts:15]: i18n

File: src/screens/discover/components/DiscoverSearch.js [hunk 2]
```diff
@@ -144,6 +148,35 @@ export default function DiscoverSearch() {
     checkAndHandlePoaps(searchQueryForPoap);
   }, [searchQueryForPoap]);
 
+  useEffect(() => {
+    const checkAndHandleMint = async seachQueryForMint => {
+      if (seachQueryForMint.includes('mint.fun')) {
+        const mintdotfunURL = seachQueryForMint.split('https://mint.fun/');
+        console.log(seachQueryForMint);
```
Comment (skylarbarrera) [inline src/screens/discover/components/DiscoverSearch.js:155]: logs

File: src/screens/mints/MintSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,741 @@
...
+      const ensName = await fetchReverseRecord(address);
+      setENSName(ensName);
+    };
+
+    if (mintCollection.creator) {
+      fetchENSName(mintCollection.creator);
+    }
+  }, [mintCollection.creator]);
+
+  // estimate gas
```
Comment (skylarbarrera) [inline src/screens/mints/MintSheet.tsx:238]: nit

File: src/screens/mints/MintSheet.tsx [hunk 4]
```diff
@@ -0,0 +1,741 @@
...
+    6
+  )} 􀄯`;
+
+  const mintPriceDisplay = convertRawAmountToBalance(
+    multiply(price || '0', quantity),
+    {
+      decimals: 18,
+      symbol: 'ETH',
+    }
+    // abbreviate if amount is >= 10,000
```
Comment (skylarbarrera) [inline src/screens/mints/MintSheet.tsx:337]: nit

File: src/screens/mints/MintSheet.tsx [hunk 5]
```diff
@@ -0,0 +1,741 @@
...
+                      horizontalPadding={0}
+                      currentNetwork={currentNetwork}
+                      theme={'dark'}
+                      marginBottom={0}
+                    />
+                  </Box>
+                </Stack>
+
+                {/*
+                          nPress={()=> ethereumUtils.openAddressInBlockExplorer(mintCollection?.contract, currentNetwork)} value={contractAddressDisplay} color={imageColor}
```
Comment (skylarbarrera) [inline src/screens/mints/MintSheet.tsx:612]: deleet

**Review by ibrahimtaveras00 — APPROVED** [review]
All issues have been fixed 👍🏽

**Review by dereknelson — APPROVED** [review]
ngl this is looking pretty mergeable 🫦

File: src/analytics/event.ts [hunk 6]
```diff
@@ -78,9 +79,14 @@ export const event = {
   nftOffersSelectedSortCriterion: 'Selected NFT Offers Sort Criterion',
   nftOffersAcceptedOffer: 'Accepted NFT Offer',
 
+  nftMintsOpenedSheet: 'Opened NFT Mint Sheet',
+  nftMintsMintingNFT: 'Minting NFT',
+  nftMintsMintedNFT: 'Minted NFT',
```
Comment (benisgold) [inline src/analytics/event.ts:84]: not a big deal but i think the naming convention should match the existing ones that i created for mints (see line 93 and below). could be confusing for our amplitude users

File: src/screens/mints/MintSheet.tsx [hunk 7]
```diff
@@ -0,0 +1,861 @@
...
+
+  const imageColor =
+    usePersistentDominantColorFromImage(imageUrl) ?? colors.paleBlue;
+
+  const sheetRef = useRef();
+  const yPosition = useSharedValue(0);
+
+  useFocusEffect(() => {
+    if (mintCollection.name && mintCollection.id) {
+      analyticsV2.track(event.nftMintsOpenedSheet, {
```
Comment (benisgold) [inline src/screens/mints/MintSheet.tsx:253]: i think in general we should decide if we want to trigger events before or after navigation occurs. like do we trigger it when the button is pressed that navs to the sheet, or when the sheet is rendered?

Comment (benisgold) [reply src/screens/mints/MintSheet.tsx:253]: just a random thought

File: src/screens/mints/MintSheet.tsx [hunk 8]
```diff
@@ -0,0 +1,861 @@
...
+        (
+          await ethereumUtils.getNativeAssetForNetwork(
+            currentNetwork,
+            accountAddress
+          )
+        )?.balance?.amount ?? 0;
+      const txFee = getTotalGasPrice();
+      const totalMintPrice = multiply(price.amount, quantity);
+      // gas price + mint price
+      // TODO: need to double check this when there are paid mints available
```
Comment (benisgold) [inline src/screens/mints/MintSheet.tsx:274]: bumping this in case its relevant

File: src/screens/mints/components/QuantityButton.tsx [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+import Row from '@/components/layout/Row';
+import { Box, Inline, Text } from '@/design-system';
+import { useTheme } from '@/theme';
+
+const PLUS_ACTION_TYPE = 'plus';
+const MINUS_ACTION_TYPE = 'minus';
+const LONG_PRESS_DELAY_THRESHOLD = 69;
+const MIN_LONG_PRESS_DELAY_THRESHOLD = 200;
+
+const Wrapper = styled(Row)({});
```
Comment (benisgold) [inline src/screens/mints/components/QuantityButton.tsx:16]: does `styled` do anything here

File: src/screens/mints/MintSheet.tsx [hunk 10]
```diff
@@ -0,0 +1,861 @@
...
+                      <Stack space="2px">
+                        {
+                          <Text
+                            color="labelSecondary"
+                            align="center"
+                            size="13pt"
+                            weight="semibold"
+                          >
+                            {quantity === Number(maxMintsPerWallet)
+                              ? 'Max'
```
Comment (benisgold) [inline src/screens/mints/MintSheet.tsx:693]: i18n

File: src/screens/mints/MintSheet.tsx [hunk 11]
```diff
@@ -0,0 +1,861 @@
...
+                    symbol="􀐾"
+                    label={i18n.t(i18n.l.minting.total_minted)}
+                    value={
+                      <Text
+                        color="labelSecondary"
+                        align="right"
+                        size="17pt"
+                        weight="medium"
+                      >
+                        {`${mintCollection.tokenCount} NFTs`}
```
Comment (benisgold) [inline src/screens/mints/MintSheet.tsx:764]: maybe i18n? pluralization might be different in different langs

File: src/screens/mints/MintSheet.tsx [hunk 12]
```diff
@@ -0,0 +1,861 @@
...
+      Network.mainnet,
+      Network.optimism,
+      Network.base,
+      Network.zora,
+    ];
+    if (!MintDotFunNetworks.includes(network)) {
+      // show alert mint.fun does not support
+      // i18n
+      // this isnt possible with our current entry points
+      Alert.alert('Mint.fun does not support this network');
```
Comment (benisgold) [inline src/screens/mints/MintSheet.tsx:395]: i18n

File: src/analytics/event.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (skylarbarrera) [reply src/analytics/event.ts:84]: I like it

File: src/hooks/useGas.ts [hunk 13]
```diff
@@ -186,6 +186,18 @@ export default function useGas({
     [dispatch]
   );
 
+  const getTotalGasPrice = () => {
```
[the remark this answers is not in the document]
Comment (skylarbarrera) [reply src/hooks/useGas.ts:189]: yess ty king

File: src/resources/nfts/simplehash/utils.ts [hunk 14]
```diff
@@ -435,7 +435,7 @@ export function simpleHashNFTToInternalNFT(nft: ValidatedSimpleHashNFT): NFT {
       externalUrl: collection.external_url ?? undefined,
       floorPrices,
       imageUrl: collection.image_url
-        ? maybeSignUri(collection.image_url)
+        ? maybeSignUri(collection.image_url, { w: 100 })
```
[the remark this answers is not in the document]
Comment (skylarbarrera) [reply src/resources/nfts/simplehash/utils.ts:438]: lol yeah idk wtf I was thinking here

**Pull request merged**